src/openrct2/ride/Ride.cpp

```diff
@@ -3509,8 +3509,7 @@ static void ride_shop_connected(Ride* ride)
     }
```

Comment:
So the actual source of the errors is in the above bit of code. In the original it would `&` the `entrance_directions` with `0xF` thats why you are getting the bad value coming in. Could you also add `& 0xF` to the above assignments of `entrance_directions`. That way it will remind us at some distant point in the future to refactor `Track`/`Flat` `SequenceProperties` into 2 accessible variables.

Reply:
Still keep the `rol4` though as it is an improvement of the code.